x/erc20/proposal_handler.go

```diff
@@ -73,13 +74,14 @@ func handleRegisterERC20Proposal(
 			return err
 		}
 
-		ctx.EventManager().EmitEvent(
-			sdk.NewEvent(
-				types.EventTypeRegisterERC20,
-				sdk.NewAttribute(types.AttributeKeyCosmosCoin, pair.Denom),
-				sdk.NewAttribute(types.AttributeKeyERC20Token, pair.Erc20Address),
-			),
-		)
+		err = ctx.EventManager().EmitTypedEvent(&types.EventRegisterCoin{
```

Comment:
This should be a different event  `EventRegisterERC20`.

If we want to maintain one event for both cases, we should rename it to `EventRegisterPair`